doc/network.md

```diff
@@ -0,0 +1,325 @@
+
+## Packets
+
+### Workspace-related Packets
+
+The server controls most of the requests sent by a client for workspace-related packets as it requires some permissions.
+Those packets are most likely sent only on build servers.
+
+#### `plasmid:workspace/enter`
+
+Packet sent when a player enters a workspace editor. The client should render the bounds of the map if present,
+and the contained regions, points, etc.  
+A `plasmid:workspace/regions` packet should be sent with this packet if the workspace contains regions.
+
+Direction: `C<-S`
+
+| Fields | Type       | Description                                                            |
+|:------:|:----------:|:-----------------------------------------------------------------------|
+| id     | Identifier | The identifier of the entered workspace.                               |
+| bounds | [Bounds]   | The bounds of the map, may be `[[0, 0, 0], [0, 0, 0]]` if not present. |
+| world  | Identifier | The world in which the workspace is present.                           |
+| data   | NBT Tag    | Arbitrary data assigned to the map template.                           |
+
+#### `plasmid:workspace/bounds`
+
+Sets the bounds of the workspace map. The client should render those bounds if non-null.
+A client can request bounds change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction `C<->S`
+
+| Fields | Type       | Description                                                                      |
+|:------:|:----------:|:---------------------------------------------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace.                                        |
+| bounds | [Bounds]   | The new bounds of the workspace map, `[[0, 0, 0], [0, 0, 0]]` if bounds is null. |
+
+#### `plasmid:workspace/data`
+
+Sets the associated data on the workspace map.
+A client can request data change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction: `C<->S`
+
+| Fields | Type       | Description                               |
+|:------:|:----------:|:------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace. |
+| data   | NBT Tag    | The NBT data.                             |
+
+#### `plasmid:workspace/region`
+
+Specifies a single region, the client should render a box, and a marker tag for it.
+If sent from a client, this is considered as an update request which can be rejected.
+
+Direction: `C<->S`
+
+| Fields    | Type       | Description                                                          |
+|:---------:|:----------:|:---------------------------------------------------------------------|
+| workspace | Identifier | The identifier of the workspace in which the region should be added. |
+| id        | [VarI32]   | The region runtime identifier.                                       |
```

Comment:
Might be worth also qualifying these as `runtime_id` or `network_id` specifically